dumpling/dumpling.go

```diff
@@ -266,3 +268,17 @@ func (m *Dumpling) constructArgs() (*export.Config, error) {
 
 	return dumpConfig, nil
 }
+
+// detectAnsiQuotes tries to detect ANSI_QUOTES from upstream. If success, change EnableANSIQuotes in subtask config
+func (m *Dumpling) detectAnsiQuotes() {
+	db, err := sql.Open("mysql", m.dumpConfig.GetDSN(""))
+	if err != nil {
+		return
+	}
+	defer db.Close()
+	enable, err := utils.HasAnsiQuotesMode(db)
+	if err != nil {
+		return
+	}
+	m.cfg.EnableANSIQuotes = enable
```

Comment:
How about logging `enable` here because we are changing the config (we may deprecate it in task config later)?

Reply:
OK, to warn user.

Reply:
https://github.com/pingcap/dm/pull/909/commits/e0d32d30c60ad860ce9843ec97cbce17df0f3d39